spark-plugins/src/test/java/co/cask/hydrator/plugin/spark/test/FeatureGeneratorTest.java

```diff
@@ -0,0 +1,330 @@
+    String sink = "hashing-tf-generator-sink";
+    /*
+     * source --> sparkcompute --> sink
+     */
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(new ETLStage("source", MockSource.getPlugin(source, INPUT)))
+      .addStage(new ETLStage("sparkcompute",
+                             new ETLPlugin(HashingTFFeatureGenerator.PLUGIN_NAME, SparkCompute.PLUGIN_TYPE,
+                                           ImmutableMap.of("numFeatures", "10",
+                                                           "outputColumnMapping", "body:result",
+                                                           "pattern", " "), null)))
+      .addStage(new ETLStage("sink", MockSink.getPlugin(sink)))
+      .addConnection("source", "sparkcompute")
+      .addConnection("sparkcompute", "sink")
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("SinglePhaseApp");
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(INPUT)
+        .set("offset", 1)
+        .set("body", "Hi I heard about Spark").build(),
+      StructuredRecord.builder(INPUT)
+        .set("offset", 2)
+        .set("body", "I wish Java could use case classes").build(),
+      StructuredRecord.builder(INPUT)
+        .set("offset", 3)
+        .set("body", "Logistic regression models are neat").build()
+    );
+
+    DataSetManager<Table> inputManager = getDataset(Id.Namespace.DEFAULT, source);
+    MockSource.writeInput(inputManager, input);
+
+    // manually trigger the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    DataSetManager<Table> labeledTexts = getDataset(sink);
+    List<StructuredRecord> structuredRecords = MockSink.readOutput(labeledTexts);
+    Assert.assertEquals(3, structuredRecords.size());
+    for (StructuredRecord record : structuredRecords) {
+      SparseVector actual = VectorUtils.fromRecord((StructuredRecord) (record.get("result")));
+      SparseVector expected;
+      if ((Integer) record.get("offset") == 1) {
+        expected = new SparseVector(10, new int[]{3, 6, 7, 9}, new double[]{2.0, 1.0, 1.0, 1.0});
+        Assert.assertEquals(expected, actual);
+      } else if ((Integer) record.get("offset") == 2) {
+        expected = new SparseVector(10, new int[]{2, 3, 4, 5, 6}, new double[]{1.0, 3.0, 1.0, 1.0, 1.0});
+        Assert.assertEquals(expected, actual);
+      } else {
+        expected = new SparseVector(10, new int[]{0, 2, 4, 5, 8}, new double[]{1.0, 1.0, 1.0, 1.0, 1.0});
+        Assert.assertEquals(expected, actual);
+      }
+    }
+  }
+
+  @Test
+  public void testHashingTFFeatureGeneratorWithArrayString() throws Exception {
```

Comment:
Agreed. Added SparkUtilsFeatureGenerator to test code specific to the utility methods.
Also, since the SprkUtils class is package private,thus creating the test within the package co.cask.hydrator.plugin.spark.